### PR TITLE
Add userAction.click to prevent pause/play when player is clicked

### PR DESCRIPTION
**Pull request by rberger**
## Description

Over the ages, many people have requested the ability to make it so clicking on the player does NOT toggle pause/play but all the other nice features such as still controlling the video via the control-bar and having the controls fade in and out based on pointer actions still work.

The most recent issue was
* #7123 
where @gkatsev showed what was needed to be done and requested a PR. 

This is a proposed PR to fix and possibly:
* #7429
* #4498
* #2444
* Many StackOverflow questions

The `pointer-events: none;` CSS suggestion does not do the same as it disables the show/hide of the control bar and other impacts.

## Specific Changes proposed
Add a userAction for the single click event to override the pause/play action.

## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [x] If necessary, more likely in a feature request than a bug fix
  - [x] Change has been verified in an actual browser (Chrome, Firefox, IE)
    - [x]Tested on macOS Chrome, Firefox and Safari
  - [ ] Unit Tests updated or fixed
    - I tried to write some tests, but I am not familiar with the testing tools and failed. Would be great if someone familiar with the test setup could do it.
  - [x] Docs/guides updated
    - [x]  Updated the Guide
  - [x] Example created ([starter template on JSBin](https://codepen.io/gkatsev/pen/GwZegv?editors=1000#0))
    - [x] `/sandbox/userAction-click.html.example`
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

File: docs/guides/options.md [hunk 1]
```diff
@@ -443,6 +444,40 @@ Defines the order in which Video.js techs are preferred. By default, this means
 
 > Type: `Object`
 
+### `userActions.click`
```
Comment (gkatsev) [inline docs/guides/options.md:447]: Thanks for updating the documentation!

File: src/js/player.js [hunk 2]
```diff
@@ -1934,10 +1934,26 @@ class Player extends Component {
...
 
-    if (this.paused()) {
-      silencePromise(this.play());
-    } else {
-      this.pause();
+    if (
+      this.options_ === undefined ||
+        this.options_.userActions === undefined ||
+        this.options_.userActions.click === undefined ||
+        this.options_.userActions.click !== false
```
Comment (gkatsev) [inline src/js/player.js:1941]: nit: spacing
```suggestion
      this.options_ === undefined ||
      this.options_.userActions === undefined ||
      this.options_.userActions.click === undefined ||
      this.options_.userActions.click !== false
```

File: src/js/player.js [hunk 3]
```diff
@@ -1934,10 +1934,26 @@ class Player extends Component {
...
+    ) {
+
+      if (
+        this.options_ !== undefined &&
+          this.options_.userActions !== undefined &&
+          typeof this.options_.userActions.click === 'function'
```
Comment (gkatsev) [inline src/js/player.js:1947]: nit: spacing
```suggestion
        this.options_ !== undefined &&
        this.options_.userActions !== undefined &&
        typeof this.options_.userActions.click === 'function'
```

**Review by gkatsev — COMMENTED** [review]
Thanks!

It's definitely been a well requested feature. Making it a user action makes total sense.

Are you able to add some tests? Should be in here https://github.com/videojs/video.js/blob/main/test/unit/player-user-actions.test.js#L8

Also, can you revert the package-lock.json changes?

File: src/js/player.js (hunk 3 above)
Comment (rberger) [reply src/js/player.js]: Yeah, I rarely edit  javascript and recently switched to doom emacs and don't have the auto indention working right yet.

Fixed the Spacing and reverted the package-lock.json.

I spent a couple hours trying to write the unit tests and it wasn't working out. Like I said, I don't normally write javascript and am not familiar with the test framework. And unfortunately this was already some deep Yak shaving by the Bike Shed at the end of a big detour trying to get contorted aspect ratios and css layout working on my main task that I'm already weeks behind on.

Maybe there is someone lurking who is familiar with the test framework could add that?  Conceptually it should be a trivial variation on the userAction.doubeClick tests. But I was not having any luck with that.

Comment (gkatsev) [reply src/js/player.js]: I'll see if I can get a few minutes to write a test or two tomorrow.

**Review by gkatsev — APPROVED** [review]

**Review by misteroneill — APPROVED** [review]

**Pull request merged**